### PR TITLE
Add `flattened` Jinja filter to debops__tpl_macros.js

### PR DESCRIPTION
**Pull request by ypid**
Opened to keep track of PR which make use of this. I would propose to wait and merge all at once when this has been properly tested.

### CODE REVIEW

File: templates/debops__tpl_macros.j2 [hunk 1]
```diff
@@ -129,3 +129,64 @@
...
 {{ merged_dict | to_json }}
 {% endmacro %}{# ]]] #}
+
+{% macro flattened() %}{# [[[ #}
+{# This macro does what the flattened lookup from Ansible fails to do in Jinja templates as of Ansible 2.2.
+## All macro arguments are flattened into one "flat" list.
+## Uses a less known feature of Jinja for using the *args and *kwargs syntax as known from
+## Python. Even if the macro does not declare any arguments, it will happyily
+## flatten any non-key-value arguments you provide.
+## Additional key-value arguments can be used to influence the behavoir of the macro.
```
Comment (drybjed) [inline templates/debops__tpl_macros.j2]: "behavoir" -> "behaviour"

Comment (ypid) [inline templates/debops__tpl_macros.j2]: That happens when my Vim does not have spell checking enabled for the current filetype and section for some reason :)

**Comment by drybjed** [comment]
@ypid Just got a thought - maybe we are doing this all wrong and we should do Ansible filter plugins instead? What do you think?

**Comment by ypid** [comment]
That is a valid point :+1: I guess the macros where just a natural starting point since we already use Jinja. Will need to be looked at.

**Pull request merged**